src/Arcus.Security.Tests.Unit/KeyVault/KeyVaultSecretProviderTests.cs

```diff
@@ -25,38 +31,71 @@ public void KeyVaultSecretProvider_CreateWithHttpScheme_ShouldFailWithUriFormatE
             string uri = $"http://{Guid.NewGuid():N}.vault.azure.net/";
 
             // Act & Assert
-            Assert.ThrowsAny<UriFormatException>(() => new KeyVaultSecretProvider(null, new KeyVaultConfiguration(uri)));
+            Assert.ThrowsAny<UriFormatException>(() => new KeyVaultSecretProvider((IKeyVaultAuthentication) null, new KeyVaultConfiguration(uri)));
         }
 
         [Fact]
         public void KeyVaultSecretProvider_CreateWithoutUri_ShouldFailWithArgumentException()
         {
-            // Arrange
-            string uri = null;
-
             // Act & Assert
-            Assert.ThrowsAny<ArgumentException>(() => new KeyVaultSecretProvider(new AzureKeyVaultAuthenticatorDummy(), new KeyVaultConfiguration(uri)));
+            Assert.ThrowsAny<ArgumentException>(
+                () => new KeyVaultSecretProvider(
+                    new AzureKeyVaultAuthenticatorDummy(), 
+                    new KeyVaultConfiguration(rawVaultUri: null)));
         }
 
         [Fact]
         public void KeyVaultSecretProvider_CreateWithoutClientFactory_ShouldFailWithArgumentException()
         {
             // Arrange
-            string uri = $"https://{Guid.NewGuid():N}.vault.azure.net/";
+            string uri = GenerateVaultUri();
```

Comment:
Let's rename to `GenerateBogusVaultUri` maybe?

Reply:
Is it actualy a generated bogus? We're not explicitly testing URI parsing and/or input validation; so I guess it would make it less obvious?

Reply:
I'm using bogus as it does not exist, not perse a reference to the libarary. Feels like you'd expect a valid vault URI here imo, but was just an idea

Reply:
:)